router/utils/types.go

```diff
@@ -0,0 +1,39 @@
+package utils
+
+import (
+	"reflect"
+	"strings"
+)
+
+// GetType returns reflect.Type of the generic parameter it receives.
+func GetType[T any]() reflect.Type { return reflect.TypeOf(new(T)).Elem() }
+
+// Nil represents an empty type.
+// You can use it with the HandlerFunc generic parameters to declare no Request with no request body, no path or query
+// params, or responses with no response body.
+type Nil *uintptr
+
+// NilType represent the type of Nil.
+var NilType = GetType[Nil]()
+
+const ignoreFieldTagValue = "-"
+
+func StructKeys(structType reflect.Type, tag string) map[string]reflect.StructField {
+	if structType == nil || structType == NilType {
+		return map[string]reflect.StructField{}
+	}
+	return FromEntries(ConcatSlices(Map(Filter(reflect.VisibleFields(structType), func(field reflect.StructField) bool {
+		return !field.Anonymous && field.IsExported() && field.Tag.Get(tag) != ignoreFieldTagValue
+	}), func(field reflect.StructField) Entry[string, reflect.StructField] {
+		name := field.Tag.Get(tag)
+		name, _, _ = strings.Cut(name, ",")
+		if name == "" {
+			name = field.Name
+		}
+		return Entry[string, reflect.StructField]{Key: name, Value: field}
+	}), ConcatSlices(Map(Filter(reflect.VisibleFields(structType), func(field reflect.StructField) bool {
+		return field.Anonymous && field.IsExported() && field.Tag.Get(tag) != ignoreFieldTagValue
+	}), func(field reflect.StructField) []Entry[string, reflect.StructField] {
+		return Entries(StructKeys(field.Type, tag))
+	})...)))
+}
```

Comment:
I think this would be clearer: 

```suggestion
func StructKeys(structType reflect.Type, tag string) map[string]reflect.StructField {
	if structType == nil || structType == NilType {
		return map[string]reflect.StructField{}
	}

	visibleFields := reflect.VisibleFields(structType)

	exportedFields := Filter(
		visibleFields,
		func(field reflect.StructField) bool {
			return field.IsExported() && field.Tag.Get(tag) != ignoreFieldTagValue
		})

	nonAnonFields := Filter(
		exportedFields,
		func(field reflect.StructField) bool { return !field.Anonymous })

	anonFields := Filter(
		exportedFields,
		func(field reflect.StructField) bool { return field.Anonymous })

	nonAnonEntries := Map(
		nonAnonFields,
		func(field reflect.StructField) Entry[string, reflect.StructField] {
			name := extractFieldName(field, tag)
			return Entry[string, reflect.StructField]{Key: name, Value: field}
		})

	anonEntries := ConcatSlices(
		Map(
			anonFields,
			func(field reflect.StructField) []Entry[string, reflect.StructField] {
				return Entries(StructKeys(field.Type, tag))
			})...)

	return FromEntries(ConcatSlices(nonAnonEntries, anonEntries))
}

func extractFieldName(field reflect.StructField, tag string) string {
	name := field.Tag.Get(tag)
	name, _, _ = strings.Cut(name, ",")
	if name == "" {
		name = field.Name
	}
	return name
}

```